index.js

```diff
@@ -80,27 +80,26 @@ module.exports = class ServeDrive extends ReadyResource {
     const isHEAD = req.method === 'HEAD'
 
     if (this._onfilter && !this._onfilter(id, filename)) {
-      res.writeHead(404).end('ENOENT')
+      res.writeHead(404)
+      res.end()
       return
     }
 
     let entry
     try {
       entry = await snapshot.entry(filename)
     } catch (e) {
-      const msg = e.code || e.message
-
-      if (e.code === 'SNAPSHOT_NOT_AVAILABLE') res.writeHead(404)
-      else {
-        this.emit('request-error', e)
-        res.writeHead(500)
+      if (e.code === 'SNAPSHOT_NOT_AVAILABLE') {
+        res.writeHead(404)
+        res.end()
+        return
       }
-      res.end(msg)
-      return
+      throw e // bubble it up
```

Comment:
Why not ignoring the error and using `request-error`? What about that `snapshot.entry` having a 30-60s timeout?

Reply:
Maybe it should emit `error` here, so it has the chance to noop it

Reply:
this is using request-error, (catch elsewhere) just fixes a reentry bug

Reply:
to your other question. not your place (as serve-drive) to specify timeouts. thats the job of the person providing the drive if they want that.

Reply:
Ah this is `_driveToRequest` which is already catched. It wasn't just the timeout but like other errors like session closed, etc but again already catched

Reply:
Ya exactly